telegram/_paidmedia.py

```diff
@@ -0,0 +1,290 @@
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program. If not, see [http://www.gnu.org/licenses/].
+"""This module contains objects that represent paid media in Telegram."""
+
+from typing import TYPE_CHECKING, Dict, Final, Optional, Sequence, Tuple, Type
+
+from telegram import constants
+from telegram._files.photosize import PhotoSize
+from telegram._files.video import Video
+from telegram._telegramobject import TelegramObject
+from telegram._utils import enum
+from telegram._utils.argumentparsing import parse_sequence_arg
+from telegram._utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class PaidMedia(TelegramObject):
+    """Describes the paid media added to a message. Currently, it can be one of:
+
+    * :class:`telegram.PaidMediaPreview`
+    * :class:`telegram.PaidMediaPhoto`
+    * :class:`telegram.PaidMediaVideo`
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`type` are equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        type (:obj:`str`): Type of the paid media.
+
+    Attributes:
+        type (:obj:`str`): Type of the paid media.
+    """
+
+    __slots__ = ("type",)
+
+    PREVIEW: Final[str] = constants.PaidMediaType.PREVIEW
+    """:const:`telegram.constants.PaidMediaType.PREVIEW`"""
+    PHOTO: Final[str] = constants.PaidMediaType.PHOTO
+    """:const:`telegram.constants.PaidMediaType.PHOTO`"""
+    VIDEO: Final[str] = constants.PaidMediaType.VIDEO
+    """:const:`telegram.constants.PaidMediaType.VIDEO`"""
+
+    def __init__(
+        self,
+        type: str,  # pylint: disable=redefined-builtin
+        *,
+        api_kwargs: Optional[JSONDict] = None,
+    ) -> None:
+        super().__init__(api_kwargs=api_kwargs)
+        self.type: str = enum.get_member(constants.PaidMediaType, type, type)
+
+        self._id_attrs = (self.type,)
+        self._freeze()
+
+    @classmethod
+    def de_json(
+        cls, data: Optional[JSONDict], bot: Optional["Bot"] = None
+    ) -> Optional["PaidMedia"]:
+        """Converts JSON data to the appropriate :class:`PaidMedia` object, i.e. takes
+        care of selecting the correct subclass.
+
+        Args:
+            data (Dict[:obj:`str`, ...]): The JSON data.
+            bot (:class:`telegram.Bot`, optional): The bot associated with this object.
+
+        Returns:
+            The Telegram object.
+
+        """
+        data = cls._parse_data(data)
```

Comment:
just an idea: we use this same logic now in several classes. We could think of extracting this logic to a helper method, possibly a protected method in TelegramObject.

Reply:
we still have to specify the `_class_mapping` on every call isn't it? which parts / how exactly do you want to extract this?

Reply:
I was thinking something like

```python
def de_json(…):
    super()._de_json_subclasses(PaidMedia, {cls.PREVIEW: …})

def _de_json_subclasses(cls, data, base_class, class_mapping):
    data = cls._parse_data(data)
     if data is None:
            return None

        if not data and cls is base_class:
            return None

        class_mapping: Dict[str, Type[PaidMedia]] = {
            cls.PREVIEW: PaidMediaPreview,
            cls.PHOTO: PaidMediaPhoto,
            cls.VIDEO: PaidMediaVideo,
        }

        if cls is PaidMedia and data.get("type") in class_mapping:
            return class_mapping[data.pop("type")].de_json(data=data, bot=bot)

        return super().de_json(data=data, bot=bot)
```

But just a thought - doesn't have to be in this PR or at all

Reply:
yeah, probably better to do in a different PR so we can apply to other classes

Reply:
if you have the time, I would be glad to see an issue for it 😅